### PR TITLE
fix: correctly handle cases where we don't use nonce2

### PR DESCRIPTION
**Pull request by obiyankenobi**

### CODE REVIEW

File: util.c [hunk 1]
```diff
@@ -2314,7 +2314,11 @@ static bool parse_notify(struct pool *pool, json_t *val)
 	pool->nonce2 = 0;
 	pool->nonce2_offset = 64;
 	if (nonce_size <= 4) {
-		pool->n2size = 0;
+		// if the nonce is shorter than 4 bytes, n2 should be zero. However, we don't wan't that
+		// as it would cause problems later, trying to copy 0 bytes.
```
Comment (jansegre) [inline util.c]: I don't think that copying 0 bytes is a problem. If I understood correctly it had more to do with generating "sub-jobs" with different nonces for each mining thread, or something like that.

I'm not objecting to the change though, only the explanation doesn't seem convincing/accurate.

**Review by jansegre — APPROVED** [review]

File: util.c (hunk 1 above)
Comment (obiyankenobi) [reply util.c]: That's true, I think the main gain here would be not filling the logs.

Also, by using `n2size = 1` and `nonce2_offset = 76`, we change the 1st byte of the 4-byte nonce. As we don't know how the asic increments the nonce, we're not sure this gives us performance gains, but it won't be worst than `n2size = 0`, as all asic chips would start working on the same nonce.

I'll update the comments.

Comment (msbrogli) [reply util.c]: I agree with @obiyankenobi. We're killing two birds with one stone, even though we're not sure if nonce2 will give us a better performance

**Review by msbrogli — APPROVED** [review]

File: util.c (hunk 1 above)
Comment (obiyankenobi) [reply util.c]: Improved a lot the comment to explain this.

**Pull request merged**